### PR TITLE
Add CompositeBoundField for improved form handling

### PR DESCRIPTION
**Pull request by timheap**
You can now do `{{ form.composite_field.child_field }}` to access the
child fields of the composite field in the template, to render them in
what ever arrangement you want.

`{{ form.composite_field.id_for_label }}` returns the ID for the first
field in the composite field.

Fields stay populated when a validation error occurs.

### CODE REVIEW

File: postgres_composite_types/forms.py [hunk 1]
```diff
@@ -32,15 +32,47 @@
...
+        <label for="{{ form.address.suburb }}">Suburb:</label>
+        {{ form.address.suburb }}
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._bound_fields_cache = {}
+
+        initial = self.form.initial.get(self.name, self.field.initial)
+        if isinstance(initial, CompositeType):
+            initial = initial.__to_dict__()
```
Comment (danni) [inline postgres_composite_types/forms.py]: Empty line after block :)

File: postgres_composite_types/forms.py [hunk 2]
```diff
@@ -32,15 +32,47 @@
...
+        data = self.form.data if self.form.is_bound else None
```
Comment (danni) [inline postgres_composite_types/forms.py]: Unrequired ternary operators reduce readability.

File: postgres_composite_types/forms.py [hunk 3]
```diff
@@ -32,15 +32,47 @@
...
+
+        self.composite_form = forms.Form(data=data, initial=initial,
+                                         prefix=self.name)
```
Comment (danni) [inline postgres_composite_types/forms.py]: Can you not just pass fields in?

File: postgres_composite_types/forms.py (hunk 2 above)
Comment (timheap) [inline postgres_composite_types/forms.py]: This one is required, sadly. `form.data` is set to `{}` if `None` is passed in, but passing `{}` means a very different thing to passing in `None`. I can make it an explicit `if: else:` block if you prefer.

File: postgres_composite_types/forms.py [hunk 4]
```diff
@@ -130,3 +170,14 @@ def value_from_datadict(self, data, files, name):
...
+    def id_for_label(self, id_):
+        """
+        Wrapper around the field widget's `id_for_label` method.
+        Useful, for example, for focusing on this field regardless of whether
+        it has a single widget or a MultiWidget.
+        """
+        if id_:
+            name = next(iter(self.widgets.keys()))
+            return '%s-%s' % (id_, name)
+        return id_
```
Comment (danni) [inline postgres_composite_types/forms.py]: Empty line after block please :)

File: postgres_composite_types/forms.py (hunk 2 above)
Comment (danni) [inline postgres_composite_types/forms.py]: Yes please :)

File: postgres_composite_types/forms.py (hunk 3 above)
Comment (timheap) [inline postgres_composite_types/forms.py]: Nope, it is not a supported parameter to `Form`. This surprised me as well.

**Comment by danni** [comment]
Nice approach. Only style issues here. I note that #1 regards the lack of tests :)

**Comment by timheap** [comment]
Formatting fixes pushed

**Pull request merged**